# Arcadia-Science/hifi2genome: Citations

Below are Nextflow and pipeline-specific citations you should include when using results from this pipeline.

# Nextflow citations

## [nf-core](https://pubmed.ncbi.nlm.nih.gov/32055031/)

> Ewels PA, Peltzer A, Fillinger S, Patel H, Alneberg J, Wilm A, Garcia MU, Di Tommaso P, Nahnsen S. The nf-core framework for community-curated bioinformatics pipelines. (2020). [doi: 10.1038/s41587-020-0439-x](https://doi.org/10.1038/s41587-020-0439-x)

## [Nextflow](https://pubmed.ncbi.nlm.nih.gov/28398311/)

> Di Tommaso P, Chatzou M, Floden EW, Barja PP, Palumbo E, Notredame C. Nextflow enables reproducible computational workflows. (2017). [doi: 10.1038/nbt.3820](https://doi.org/10.1038/nbt.3820)

## [MultiQC](https://academic.oup.com/bioinformatics/article/32/19/3047/2196507)

> Ewels P, Magnusson M, Lundin S, Kaller M. MultiQC: summarize analysis results for multiple tools and samples in a single report. (2016). [doi:10.1093/bioinformatics/btw354](https://doi.org/10.1093/bioinformatics/btw354)

# Specific Pipeline tools

## [Flye][https://www.nature.com/articles/s41587-019-0072-8]

> Kolmogorov M, Yuan J, Lin Y, Pevzner P. Assembly of long, error-prone reads using repeat graphs. (2019). [doi:10.1038/s41587-019-0072-8](https://doi.org/10.1038/s41587-019-0072-8)

## [BUSCO](https://academic.oup.com/bioinformatics/article/31/19/3210/211866)

> Simao F, Waterhouse R, Ionnidis P, Kriventseva E, Zdobnov E. BUSCO: assessing genome assembly and annotation completeness with single-copy orthologs. (2015). [doi:10.1093/bioinformatics/btv351](https://doi.org/10.1093/bioinformatics/btv351)

## [minimap2](https://academic.oup.com/bioinformatics/article/34/18/3094/4994778)

> Li H. Minimap2: pairwise alignment for nucleotide sequences. (2018). [https://doi.org/10.1093/bioinformatics/bty191](https://doi.org/10.1093/bioinformatics/bty191)

## [QUAST](https://academic.oup.com/bioinformatics/article/29/8/1072/228832?login=false)

> Gurevich A, Saveliev V, Vyahhi N, Tesler G. QUAST: quality assessment tool for genome assemblies. (2013). [doi:10.1093/bioinformatics/btt086](https://doi.org/10.1093/bioinformatics/btt086)

## [Samtools](https://academic.oup.com/bioinformatics/article/25/16/2078/204688)
> Li H, Handsaker B, Wysoker A, Fennell T, Ruan J, Homer N, Marth G, Abecasis G, Durbin R,1000 Genome Project Data PRocessing Subgroup. (2009). [https://doi.org/10.1093/bioinformatics/btp352](https://doi.org/10.1093/bioinformatics/btp352)
# Software packaging/containerisation tools

- [Docker](https://dl.acm.org/doi/10.5555/2600239.2600241)

- [Singularity](https://pubmed.ncbi.nlm.nih.gov/28494014/)
  > Kurtzer GM, Sochat V, Bauer MW. Singularity: Scientific containers for mobility of compute. (2017). [https://doi.org/10.1371/journal.pone.0177459](https://doi.org/10.1371/journal.pone.0177459)
